src/services/facilityService.js

```diff
@@ -0,0 +1,88 @@
+import db from '../models';
+import { Helpers } from '../utils';
+
+const {
+  Facility, sequelize, RoomCategory, Room, FacilityAmenity, Amenity
+} = db;
+const { extractArrayRecords, updateCollection } = Helpers;
+
+/**
+ * A collection of methods that handles the database interactions
+ * for managing a Facility as an entity of the App.
+ *
+ * @class FacilityService
+ */
+class FacilityService {
+  /**
+   * Checks the room category value and creates a new record for any category added
+   * by user.
+   * @static
+   * @param {array} rooms - An array of rooms of a specific facility.
+   * @returns {Promise<Array>} A promise object with updated room properties.
+   * @memberof FacilityService
+   */
+  static async sortRoomCategory(rooms) {
+    const updatedRooms = rooms.map(async (room) => {
+      const { label } = await RoomCategory.findByPk(room.roomCategoryId);
+      if (label === 'Others') {
+        const { id } = await RoomCategory.create({ description: ' ', label: room.newCategory });
+        room.roomCategoryId = id;
+      }
+      return room;
+    });
+    return Promise.all(updatedRooms);
+  }
+
+  /**
+   * Creates an array of amenities for a specific facility.
+   * @static
+   * @param {object} amenities - Facility data to be recorded in the database.
+   * @param {number} id - Facility id.
+   * @returns {array}  An array of facility amenities.
+   * @memberof FacilityService
+   */
+  static sortFacilityAmenities(amenities, id) {
+    return amenities.map((item) => ({ amenityId: item, facilityId: id }));
```

Comment:
Here's something I noticed while writing my implementation; since amenities isn't marked as a required field on the validation middleware, in the case where nothing is passed as amenities, this code throws an array while trying to loop through nothing. My workaround was to check if the amenities field is true before looping like so: `return amenities ? amenities.map((item) => ({ amenityId: item, facilityId: id})) : [];` you could as well verify that amenities is not false before calling the method in the first place

Reply:
done @King-Success